builtin/index-pack.c

```diff
@@ -461,8 +461,12 @@ static void *unpack_entry_data(off_t offset, size_t size,
 			stream.avail_out = sizeof(fixed_buf);
 		}
 	} while (status == Z_OK);
-	if (stream.total_out != size || status != Z_STREAM_END)
-		bad_object(offset, _("inflate returned %d"), status);
+	if (stream.total_out != size)
+		// BUGS OUT HERE
+		bad_object(offset, _("stream.total_out != size: inflate returned %d"), status);
+	if (status != Z_STREAM_END)
+		// BUGS OUT HERE
+		bad_object(offset, _("status != Z_STREAM_END: inflate returned %d"), status);
```

Comment:
I do agree with this change (but highly recommend removing the C++-style comment...).

Also, the commit message's oneline should be improved. Maybe "index-pack: make error messages upon failed inflate() more helpful"?

Reply:
Comment - yes  - propagated from the original tests.

The error message was simply the condition so I could see which it was (2 very similar ones adjacent.
I deliberately separated the tests so that the different fails could be seen. - too many opportunities for failure otherwise.